addons/hr_contract/models/hr_contract.py

```diff
@@ -95,23 +95,18 @@ def _compute_employee_contract(self):
             contract.resource_calendar_id = contract.employee_id.resource_calendar_id
             contract.company_id = contract.employee_id.company_id
 
-    @api.onchange('company_id')
-    def _onchange_company_id(self):
-        if self.company_id:
-            structure_types = self.env['hr.payroll.structure.type'].search([
-                '|',
-                ('country_id', '=', self.company_id.country_id.id),
-                ('country_id', '=', False)])
-            if structure_types:
-                self.structure_type_id = structure_types[0]
-            elif self.structure_type_id not in structure_types:
-                self.structure_type_id = False
+    @api.depends('company_id')
+    def _compute_structure_calendar(self):
+        StructureType = self.env['hr.payroll.structure.type']
```

Comment:
```suggestion
        structure_type = self.env['hr.payroll.structure.type']
```

Reply:
this is to signify that I'm using this as the class, not the instance.